README.md
# Simple Social Share

Simple Social Share is a simple way of creating social share buttons on your website without loading all the third party scripts from these networks. It is available in plain javascript and as jQuery plugin. It is based on a blog by Jonathan Suh: [Responsible Social Share Links](https://jonsuh.com/blog/social-share-links/).

Version 0.4.0

For changelog see: https://github.com/MightyMedia/Simple-Social-Share/blob/master/CHANGELOG.md

The jQuery plugin requires jQuery 1.7 or newer.

Licensed under:
MIT License - https://github.com/MightyMedia/Simple-Social-Share/blob/master/LICENSE

## Requirements

* jQuery 1.7+ (only when using the jQuery plugin)

## Installation
To use Simple Social Share make sure you have jQuery 1.7 or newer. Next, just include the plugin file and you are ready to go.

__Plain javascript__
```html
<script type="text/javascript" src="simpleSocialShare.min.js"></script>
```

__jQuery__
```html
<script type="text/javascript" src="jquery.simpleSocialShare.min.js"></script>
```

## Basic usage

### Initialize

__Plain javascript__
```javascript
// Without any input, every element with the class simpleSocialShare will get the magic.
simpleSocialShare.init();

// With selector, can be #id, .className or tagName
simpleSocialShare.init('.socialShareBtn');

// With elements, can be 1 DOM element, NodeList or HTMLCollection
simpleSocialShare.init(document.getElementsByClassName('socialShareBtn'));
```

__jQuery__
```javascript
$(document).ready(function(){
    $('.socialShareBtn').simpleSocialShare();
});
```

Now every element with the class `socialShareBtn` will turn in to a functional share link.

```html
<a href="#" class="socialShareBtn" data-share-url="http://mightymedia.nl" data-share-network="twitter" data-share-text="Share this awesome link on Twitter">Share on Twitter</a>
```

### Initialize with Options

There are several basic options you can pass along during the initialization. They can be overwritten by individual data attributes per network.

__Plain javascript__
```javascript
// Initialization with before and after callback
simpleSocialShare.init( '.socialShareBtn', {

	before	: 'callbackFunctionBefore',
	after	: 'callbackFunctionAfter'

} );
```

__Overwrite or set individually by data attribute__
```html
<a href="#" class="socialShareBtn" data-share-url="http://mightymedia.nl" data-share-network="twitter" data-share-text="Share this awesome link on Twitter" data-share-before="callbackFunctionBefore" data-share-after="callbackFunctionAfter">Share on Twitter</a>
```
## Initialize Options

The options you can pass along during initialization:

| Option        | Description                                                               | Notes    |
| ------------- | ------------------------------------------------------------------------- | -------- |
| before        | Callback function to call before the popup opens                          | optional |
| after         | Callback function to call after the popup opened                          | optional |


### Destroy

To completely remove all added Simple Social Share functionality:

__Plain javascript__
```javascript
// Input can be the same as used for simpleSocialShare.init();
simpleSocialShare.destroy('selector');
```

__jQuery__
```javascript
$(document).ready(function(){
    $('.socialShareBtn').simpleSocialShare('destroy');
});
```

## Data Attributes Options

The options for a share link/button are set via data attributes on the element. Below is a list with the options.

| Option        | Description                                | Network                      | Notes    |
| ------------- | ------------------------------------------ | ---------------------------- | -------- |
| share-network | The network to share on                    | all networks                 | required |
| share-url     | The URL to be shared                       | all networks                 | required |
| share-text    | The text to pass along with the URL        | linkedin, pinterest, twitter | optional |
| share-title   | The title to add the the share             | linkedin only                | optional |
| share-via     | Account name to mention in the share/tweet | twitter only                 | optional |
| share-tags    | Tags to include to the share               | pinterest, twitter           | optional |
| share-media   | Image to add to the share                  | pinterest only               | optional |
| share-before  | Callback function before the popup opens   | all networks                 | optional |
| share-after   | Callback function after the popup opened   | all networks                 | optional |


## Issues

If you have any ideas or bugs, please submit them to the GitHub issue tracker at https://github.com/MightyMedia/Simple-Social-Share/issues.


CHANGELOG.md
# Changelog

## [0.4.0] - 2020-11-14

### Added
* Added before and after callbacks

### Changed
* Changed Changelog format
* Removed Google+ from demo


## [0.3.0] - 2018-09-06

### Added
* Added non-jQuery version
* Added gulp for building and compiling


## [0.2.0] - 2016-11-23

### Added
* Added destroy option
* Added logging instead of trowing an error when it makes more sense

### Changed
* Updated from codekit 2 to 3 for compiling
* Updated to jQuery 3.1.1 in demo


## [0.1.2] - 2016-08-30

### Added
* Added package.json for NPM

### Fixed
* Fixed some typos

### Changed
* Updated documentation


## [0.1.1] - 2016-08-23

### Added
* Added documentation

### Changed
* Some code cleanup


## [0.1.0] - 2016-08-23

* Initial public beta release
* Updates to travis config
* Added CHANGELOG
* Updated README


gulpfile.js
var gulp = require('gulp'),
    concat = require('gulp-concat'),
    uglify = require('gulp-uglify'),
    rename = require('gulp-rename'),
    jshint = require('gulp-jshint'),
    copy = require('gulp-copy'),
    combiner = require('stream-combiner2');

const paths = {
	plain: {
		src: 'simpleSocialShare.js',
		dest: './'
	},
	jquery: {
		src: 'jquery.simpleSocialShare.js',
		dest: './'
    }
}

// Compile jQuery
gulp.task('compileJquery', () => {

    var combined = combiner.obj([

        gulp.src(paths.jquery.src)
            .pipe(jshint({
                "strict": true,
                "scripturl": true,
                "devel": true,
                "curly": true,
                "undef": true,
                "unused": true,
                "eqeqeq": true,
                "eqnull": true,
                "browser": true,
                "camelcase": false,
                "esversion": 6,
                "jquery": true,
                "globals": {
                    "linkMt": true
                }
            }))
            .pipe(jshint.reporter('jshint-stylish'))
            .pipe(jshint().on('error', errorAlert))
            //.pipe(concat('app.js'))
            //.pipe(gulp.dest(paths.plain.dest))
            .pipe(uglify().on('error', errorAlert))
            .pipe(rename('jquery.simpleSocialShare.min.js'))
            .pipe(gulp.dest(paths.jquery.dest))

    ]);

    combined.on('error', console.error.bind(console));

    return combined;
});

// Compile plain
gulp.task('compilePlain', () => {

    var combined = combiner.obj([

        gulp.src(paths.plain.src)
            .pipe(jshint({
                "strict": true,
                "scripturl": true,
                "devel": true,
                "curly": true,
                "undef": true,
                "unused": true,
                "eqeqeq": true,
                "eqnull": true,
                "browser": true,
                "camelcase": false,
                "esversion": 6,
                "jquery": false,
                "globals": {
                    "linkMt": true
                }
            }))
            .pipe(jshint.reporter('jshint-stylish'))
            .pipe(jshint().on('error', errorAlert))
            //.pipe(concat('app.js'))
            //.pipe(gulp.dest(paths.plain.dest))
            .pipe(uglify().on('error', errorAlert))
            .pipe(rename('simpleSocialShare.min.js'))
            .pipe(gulp.dest(paths.plain.dest))

    ]);

    combined.on('error', console.error.bind(console));

    return combined;
});

function errorAlert(err) {
    console.log(err.toString());
    this.emit("end");
}

// Watch Files For Changes
gulp.task('watch', function() {
    gulp.watch(paths.plain.src, gulp.series('compilePlain'));
    gulp.watch(paths.jquery.src, gulp.series('compileJquery'));
});

gulp.task('default', gulp.series('compilePlain', 'compileJquery'));


simpleSocialShare.js
/* exported simpleSocialShare */

// Utility for creating objects in older browsers
if (typeof Object.create !== 'function') {
    Object.create = function(obj) {
        "use strict";
        function F() {}
        F.prototype = obj;
        return new F();
    };
}

// Create event listeners with jQuery like namespacing
var eventsNamespaceBinder = {
    on: function(event, cb, opts) {
        "use strict";

        if (!this.namespaces) { // save the namespaces on the DOM element itself
            this.namespaces = {};
        }

        this.namespaces[event] = cb;
        var options = opts || false;

        this.addEventListener(event.split('.')[0], cb, options);
        return this;
    },

    off: function(event) {
        "use strict";

        if(typeof this.namespaces !== 'undefined') {
            if(event.split('.')[1] === undefined) {
                for (var key in this.namespaces) {
                    if (key.split('.')[0] === event) {
                        this.removeEventListener(event, this.namespaces[key]);
                        delete this.namespaces[key];
                    }
                }
            } else {
                if (this.namespaces.hasOwnProperty(event)) {
                    this.removeEventListener( event.split('.')[0], this.namespaces[event]);
                    delete this.namespaces[event];
                }
            }
        }

        return this;
    }
};

// Extend the DOM with these above custom methods
window.on = Element.prototype.on = eventsNamespaceBinder.on;
window.off = Element.prototype.off = eventsNamespaceBinder.off;


/*!
 * Simple Social Share
 *
 * Simple Social Share is a simple way of creating social share buttons on your website without loading all the third party scripts from these networks.
 *
 * Based on a blog by Jonathan Suh: https://jonsuh.com/blog/social-share-links/
 *
 * Version 0.3.0
 * Licensed under the MIT License: https://github.com/MightyMedia/Simple-Social-Share/blob/master/LICENSE
 *
 * https://github.com/MightyMedia/Simple-Social-Share
 *
 * Copyright 2016, Bas van den Wijngaard
 *
 */

var simpleSocialShare = (function(){
    "use strict";

    var pluginName = 'simpleSocialShare';
    var storageName = 'plugin_' + pluginName;
    var settings = {
        selector: '.simpleSocialShare',
        before: null,
        after: null
    };

    var tools = {
        /*
         * Default logger
         */
        log: function() {
            if (window.console && console.log) {
                console.log('[social-share] ' + Array.prototype.join.call(arguments, ' ') );
            }
        },

        /*
         * Get type of object
         *
         * @see: https://stackoverflow.com/questions/332422/get-the-name-of-an-objects-type
         */
        getType: function(object) {
            return Object.prototype.toString.call(object).slice(8, -1);
        },

        /*
         * Get type of object
         */
        extend: function() {
            var obj, name, copy, target = arguments[0] || {}, i = 1, length = arguments.length;

            for (; i < length; i++) {
                if ((obj = arguments[i]) !== null) {
                    for (name in obj) {
                        copy = obj[name];

                        if (target === copy) {
                            continue;
                        } else if (copy !== undefined) {
                            target[name] = copy;
                        }
                    }
                }
            }

            return target;
        }
    };

    var pluginObject = {

        popupwindow: function(url, title, w, h) {
            var left = (screen.width/2)-(w/2);
            var top = (screen.height/2)-(h/2);
            return window.open(url, title, 'toolbar=no, location=no, directories=no, status=no, menubar=no, scrollbars=no, resizable=no, copyhistory=no, width='+w+', height='+h+', top='+top+', left='+left);
        },

        bind: function() {
            var self = this;
            // Do your binds
            self.element.on('click' + self.namespace, function(e) {
                e.preventDefault();
                self.openSocialSharePopup(self);
            });
        },

        /*
         * Get correct setting for element
         * 
         * If setting is undefined if will be overwritten by the general settings 
         */
        getSetting: function( value, setting ) {
            if ( settings[setting] !== value && typeof value !== 'undefined' ) {
                return value;
            } else {
                return settings[setting];
            }
        },

        openSocialSharePopup: function(self) {	

            var doShare 		= false;
            var shareUrl		= self.element.dataset.shareUrl;
            var shareNetwork	= self.element.dataset.shareNetwork;
            var shareText		= self.element.dataset.shareText;
            var shareTitle		= self.element.dataset.shareTitle;
            var shareVia		= self.element.dataset.shareVia;
            var shareTags		= self.element.dataset.shareTags;
            var shareMedia		= self.element.dataset.shareMedia;

            // check if before and after are set in the general settings
            var shareBefore		= self.getSetting( self.element.dataset.shareBefore, 'before' );
            var shareAfter		= self.getSetting( self.element.dataset.shareAfter, 'after' );

            var networkShareUrl	= '';

            // Check if shareUrl and shareNetwork are present and shareNetwork is in the list of allowed networks
            if (typeof shareUrl !== 'undefined' && typeof shareNetwork !== 'undefined' && self.allowedNetworks.indexOf(shareNetwork) > -1) {
                // At this point we don't automaticly fetch the current URL from the browser
                doShare = true;
            }

            if (doShare === true) {

                // do before callback
                if ( null !== shareBefore ) {
                    window[ shareBefore ]( self );
                }

                switch(shareNetwork) {
                    case 'facebook':
                        networkShareUrl = 'https://www.facebook.com/sharer/sharer.php?u=' + encodeURIComponent(shareUrl);
                        break;

                    case 'googleplus':
                        networkShareUrl = 'https://plus.google.com/share?url=' + encodeURIComponent(shareUrl);
                        break;

                    case 'linkedin':
                        networkShareUrl = 'https://www.linkedin.com/shareArticle?mini=true&url=' + encodeURIComponent(shareUrl) + '&source=' + encodeURIComponent(shareUrl);

                        if (typeof shareTitle !== 'undefined' &&  shareTitle !== '') {
                            networkShareUrl += '&title=' + encodeURIComponent(shareTitle);
                        }

                        if (typeof shareText !== 'undefined' && shareText !== '') {
                            networkShareUrl +=  '&summary=' + encodeURIComponent(shareText);
                        }

                        break;


                    case 'pinterest':
                        networkShareUrl = 'https://www.pinterest.com/pin/create/button/?url=' + encodeURIComponent(shareUrl);

                        if (typeof shareMedia !== 'undefined' && shareMedia !== '') {
                            networkShareUrl += '&media=' + encodeURIComponent(shareMedia);
                        }

                        if (typeof shareText !== 'undefined' && shareText !== '') {
                            networkShareUrl += '&description=' + encodeURIComponent(shareText);
                        }

                        if (typeof shareTags !== 'undefined' && shareTags !== '') {
                            networkShareUrl += '&hashtags=' + shareTags;
                        }

                        break;

                    case 'reddit':
                        networkShareUrl = 'http://www.reddit.com/submit/?url=' + encodeURIComponent(shareUrl);
                        break;


                    case 'twitter':
                        networkShareUrl = 'https://twitter.com/intent/tweet?&url=' + encodeURIComponent(shareUrl);

                        if (typeof shareText !== 'undefined' && shareText !== '') {
                            networkShareUrl += '&text=' + encodeURIComponent(shareText);
                        }

                        if (typeof shareVia !== 'undefined' && shareVia !== '') {
                            networkShareUrl += '&via=' + encodeURIComponent(shareVia);
                        }

                        if (typeof shareTags !== 'undefined' && shareTags !== '') {
                            networkShareUrl += '&hashtags=' + shareTags;
                        }

                        break;

                    default:
                        return false;
                }
            
                // Calculate the position of the popup so it’s centered on the screen.
                self.popupwindow(networkShareUrl, '', 500, 300);

                // do after callback
                if ( null !== shareAfter ) {
                    window[ shareAfter ]( self );
                }

            }

        },

        init: function(options,element) {
            var self = this;

            // Initialise options
            self.options = tools.extend(settings, options || {});

            // Set plugin event namespace
            self.namespace = '.' + pluginName + '.' + new Date().getTime();

            // Store current element
            self.element = element;

            // Networks this plugin is compatible with
            self.allowedNetworks = ['facebook','twitter','linkedin','googleplus', 'pinterest', 'reddit'];

            // Bind events after intialization
            self.bind();

            // Return plugin instance
            return self;
        }
    };

    var enableOnDomObject = function(object) {
        var pluginInstance = object[storageName];
        var instanceOptions = {};

        if (!pluginInstance) {
            pluginInstance = Object.create(pluginObject).init(instanceOptions, object);
            object[storageName] = pluginInstance;

        } else {
            tools.log('Simple Social Share is already initialized for this object.');
        }
    };

    var destroyOnDomObject = function(object) {
        var pluginInstance = object[storageName];

        if (!pluginInstance) {
            tools.log('Simple Social Share is not initialized for this object yet.');

        } else {
            // Delete click event
            object.off('click' + pluginInstance.namespace);

            // Delete data from dom object
            delete object[storageName];
        }
    };

    var actionBySelector = function(input, action) {
        if (null === action || action !== 'destroy') {
            action = 'enable';
        }
        var selector = '';
        var elements = {};

        if (input.indexOf('#') === 0) {
            // Get element by ID
            selector = input.substr(1);
            var element = document.getElementById(selector);

            if (null !== element) {
                if (action === 'destroy') {
                    destroyOnDomObject(element);
                } else {
                    enableOnDomObject(element);
                }
            } else {
                tools.log('No element with ID \'' + selector + '\' found.');
            }

        } else if (input.indexOf('.') === 0) {
            // Get element by class name (or use query selector)
            selector = input.substr(1);
            elements = document.getElementsByClassName(selector);

            if (null !== elements && elements.length > 0) {
                for (var i = 0; i < elements.length; i++) {
                    if (action === 'destroy') {
                        destroyOnDomObject(elements[i]);
                    } else {
                        enableOnDomObject(elements[i]);
                    }
                }
            } else {
                tools.log('No elements with class name \'' + selector + '\' found.');
            }

        } else if (input.length > 0) {
            // Get element by tag name (or use query selector)
            elements = document.getElementsByTagName(input);

            if (null !== elements && elements.length > 0) {
                for (var j = 0; j < elements.length; j++) {
                    if (action === 'destroy') {
                        destroyOnDomObject(elements[j]);
                    } else {
                        enableOnDomObject(elements[j]);
                    }
                }
            } else {
                tools.log('No elements with tag name \'' + input + '\' found.');
            }
        } else {
            tools.log('Empty selector [' + action + ']');
        }
    };

    var updateSettings = function( selector, options ) {

		if ( typeof options !== 'undefined' && options !== null ) {

			Object.keys( options ).map(function( optionKey /*, index*/ ) {
				//var value = options[optionKey];
				
				if(typeof settings[optionKey] !== 'undefined') {
					settings[optionKey] = options[optionKey];
				} else {
					tools.log( optionKey + ' is not a valid option');
				}
			
			});

		}

    };

    var destroy = function() {
        var args = Array.prototype.slice.call(arguments);

        if (null !== args && null !== args[0]) {
            var options = args[0];

            if (typeof options === 'object') {
                if (options instanceof HTMLCollection || options instanceof NodeList) {
                    // Multiple elements, loop them
                    for (var i = 0; i < options.length; i++) {
                        destroyOnDomObject(options[i]);
                    }
                } else {
                    // Probably 1 dom object
                    destroyOnDomObject(options);
                }

            } else if (typeof options === 'string') {
                // Enable by selector input
                actionBySelector(options, 'destroy');

            } else {
                // No input
                tools.log('simpleSocialShare.destroy() needs input [2]');
            }

        } else {
            tools.log('simpleSocialShare.destroy() needs input [1]');
        }
    };

    var init = function( selector, options ) {

        if (null === options || options === '') {
            options = {};
        }

        updateSettings( selector, options );

        if (typeof selector === 'object') {
            // Check if it's a NodeList or a HTMLCollection (list of elements)
            if (selector instanceof HTMLCollection || selector instanceof NodeList) {
                // Multiple elements, loop them
                for (var i = 0; i < selector.length; i++) {
                    enableOnDomObject(selector[i]);
                }
            } /*else if (tools.getType(options) === 'Object') {
                // We don't have any options yet
                // Options object
                console.log('options object');
                console.log(typeof options);
                console.log(tools.getType(options));
                console.log(options);
                //socialShare.init(options);
            }*/ else {
                // Probably 1 dom object
                enableOnDomObject(selector);
            }
        } else if (typeof selector === 'string') {
            // Enable by selector input
            actionBySelector(selector, 'enable');
        } else {
            // Use default classname selector
            actionBySelector(settings.selector, 'enable');
        }
    };

    return {
        init:init,
        destroy:destroy
    };

})();


jquery.simpleSocialShare.js
// Utility for creating objects in older browsers
if (typeof Object.create !== 'function') {
	Object.create = function(obj) {
    	"use strict";
		function F() {}
		F.prototype = obj;
		return new F();
	};
}

/*!
 * Simple Social Share for jQuery
 *
 * Simple Social Share for jQuery is a simple way of creating social share buttons on your website without loading all the third party scripts from these networks.
 *
 * Based on a blog by Jonathan Suh: https://jonsuh.com/blog/social-share-links/
 *
 * Version 0.2.0
 * Licensed under the MIT License: https://github.com/MightyMedia/Simple-Social-Share/blob/master/LICENSE
 *
 * Requires:
 * - jQuery 1.7 or higher
 *
 * https://github.com/MightyMedia/Simple-Social-Share
 *
 * Copyright 2016, Bas van den Wijngaard
 *
 */
(function($, window, document, undefined) {
    "use strict";

	var pluginName = 'simpleSocialShare';
	var storageName = 'plugin_' + pluginName;

	var pluginObject = {

		init: function(options, element) {

			var self = this;

			// Initialise options
			self.options = $.extend(true, {}, $.fn[pluginName].options, options);

			// Set plugin event namespace
			self.namespace = '.' + pluginName + '.' + new Date().getTime();

			// Store current element
			self.element = element;
			self.$element = $(element);

			// Init your plugin stuff here
			self.allowedNetworks = ['facebook','twitter','linkedin','googleplus', 'pinterest', 'reddit'];

			// Bind events after intialisation
			self.bind();

			// Return plugin instance
			return self;

		},

		bind: function() {

			var self = this;

			// Do your binds
			self.bindProxied(self.$element, 'click', self.openSocialSharePopup);

		},

		bindProxied: function($element, event, method) {

			var self = this;
			$element.on(event + self.namespace, $.proxy(function(e) {
				return method.call(self, e);
			}, self));

		},

		destroy: function() {

			var self = this;
			// Remove all binds from element
			self.$element.off(self.namespace);
			// Remove plugin instance from object
			self.$element.removeData(storageName);

		},

		openSocialSharePopup: function(e) {

			e.preventDefault();
			var self			= this;

			var doShare 		= false;
			var shareUrl		= self.$element.data('share-url');
			var shareNetwork	= self.$element.data('share-network');
			var shareText		= self.$element.data('share-text');
			var shareTitle		= self.$element.data('share-title');
			var shareVia		= self.$element.data('share-via');
			var shareTags		= self.$element.data('share-tags');
			var shareMedia		= self.$element.data('share-media');
			var networkShareUrl	= '';

			// Check if shareUrl and shareNetwork are present and shareNetwork is in the list of allowed networks
			if (typeof shareUrl !== 'undefined' && typeof shareNetwork !== 'undefined' && self.allowedNetworks.indexOf(shareNetwork) > -1) {

				// At this point we don't automaticly fetch the current URL from the browser
				doShare = true;

			}

			if (doShare === true) {

				switch(shareNetwork) {

					case 'facebook':

						networkShareUrl = 'https://www.facebook.com/sharer/sharer.php?u=' + encodeURIComponent(shareUrl);

						break;


					case 'googleplus':

						networkShareUrl = 'https://plus.google.com/share?url=' + encodeURIComponent(shareUrl);

						break;


					case 'linkedin':

						networkShareUrl = 'https://www.linkedin.com/shareArticle?mini=true&url=' + encodeURIComponent(shareUrl) + '&source=' + encodeURIComponent(shareUrl);

						if (typeof shareTitle !== 'undefined' &&  shareTitle !== '') {
							networkShareUrl += '&title=' + encodeURIComponent(shareTitle);
						}
						if (typeof shareText !== 'undefined' && shareText !== '') {
							networkShareUrl +=  '&summary=' + encodeURIComponent(shareText);
						}

						break;


					case 'pinterest':

						networkShareUrl = 'https://www.pinterest.com/pin/create/button/?url=' + encodeURIComponent(shareUrl);

						if (typeof shareMedia !== 'undefined' && shareMedia !== '') {
							networkShareUrl += '&media=' + encodeURIComponent(shareMedia);
						}
						if (typeof shareText !== 'undefined' && shareText !== '') {
							networkShareUrl += '&description=' + encodeURIComponent(shareText);
						}
						if (typeof shareTags !== 'undefined' && shareTags !== '') {
							networkShareUrl += '&hashtags=' + shareTags;
						}

						break;


					case 'reddit':

						networkShareUrl = 'http://www.reddit.com/submit/?url=' + encodeURIComponent(shareUrl);

						break;


					case 'twitter':

						networkShareUrl = 'https://twitter.com/intent/tweet?&url=' + encodeURIComponent(shareUrl);

						if (typeof shareText !== 'undefined' && shareText !== '') {
							networkShareUrl += '&text=' + encodeURIComponent(shareText);
						}
						if (typeof shareVia !== 'undefined' && shareVia !== '') {
							networkShareUrl += '&via=' + encodeURIComponent(shareVia);
						}
						if (typeof shareTags !== 'undefined' && shareTags !== '') {
							networkShareUrl += '&hashtags=' + shareTags;
						}

						break;


					default:

						return false;

				}

			}

			if (doShare) {

				// Calculate the position of the popup so it’s centered on the screen.
				self.popupwindow(networkShareUrl, '', 500, 300);

			}

		},

		popupwindow: function(url, title, w, h) {

			var left = (screen.width/2)-(w/2);
			var top = (screen.height/2)-(h/2);

			return window.open(url, title, 'toolbar=no, location=no, directories=no, status=no, menubar=no, scrollbars=no, resizable=no, copyhistory=no, width='+w+', height='+h+', top='+top+', left='+left);

		}

	};

	$.fn[pluginName] = function(options) {

		var args = Array.prototype.slice.call(arguments);

		return this.each(function() {

			var pluginInstance = $.data(this, storageName);
			var instanceOptions = {};

			if (typeof options === 'object' || options === 'init' || !options) {

				if (!pluginInstance) {

					if (options === 'init') {

						instanceOptions = args[1] || {};

					}

					pluginInstance = Object.create(pluginObject).init(instanceOptions, this);
					$.data(this, storageName, pluginInstance);

				} else {

					$.fn[pluginName].log('Simple Social Share is already initialized for this object.');
					return;

				}
			} else if (!pluginInstance) {

				$.fn[pluginName].log('Simple Social Share is not initialized for this object yet.');
				return;

			} else if (pluginInstance[options]) {

				var method = options;
				instanceOptions = args.slice(1);
				pluginInstance[method].apply(pluginInstance, instanceOptions);

			} else {

				$.error('Method ' + options + ' does not exist on jQuery.' + pluginName + '.');
				return;

			}

		});

	};

	// default logger
	$.fn[pluginName].log = function log() {

		/* global console:true */
		if (window.console && console.log) {

			console.log('[social-share] ' + Array.prototype.join.call(arguments, ' ') );

		}

	};

/*
	$.fn[pluginName].options = {
		key: 'value'
	};
*/

})(jQuery, window, document);